trickster/exceptions.py

```diff
@@ -0,0 +1,23 @@
+"""Custom exceptions for API."""
+
+
+class TricksterBaseError(Exception):
```

Comment:
I'll show you something cool

```
class TricksterBaseError(Exception):
    def __str__(self) -> str:
        if self.args:
            return self.args[0]
        else:
            return str(self.__cause__)


try:
    0 / 0
except Exception as e:
    raise BaseError() from e
```

Now you don't need the constructor at all, including the variable `detail`. And you don't even have to pass the text argument like this:

`raise ValidationError(str(exception))`

It will automatically grab it if you raise the exception properly:

`raise ValidationError() from exception`

But you can still pass different message

`raise ValidationError('some other message') from exception`

Reply:
wow, that's cool